src/puppetlabs/trapperkeeper/services/metrics/metrics_core.clj

```diff
@@ -34,24 +34,27 @@
 (def MetricsServiceContext
   {:registries (schema/atom {schema/Any RegistryContext})})
 
+(def Keyword-or-Str (schema/if keyword? schema/Keyword schema/Str))
+
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Private
 
 (schema/defn jmx-reporter :- JmxReporter
   [registry :- MetricRegistry
-   domain :- (schema/maybe schema/Str)]
+   domain :- (schema/maybe schema/Keyword)]
   (let [b (JmxReporter/forRegistry registry)]
-    (when-let [^String d domain]
-      (.inDomain b d))
+    (when domain
+      (.inDomain b (name domain)))
     (.build b)))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
 
 (schema/defn initialize :- RegistryContext
   [config :- MetricsConfig
-   domain :- (schema/maybe schema/Str)]
-  (let [jmx-config (get-in config [:reporters :jmx])
+   domain :- (schema/maybe Keyword-or-Str)]
```

Comment:
Can this be just `schema/maybe Keyword` now, or would that be an API change? Otherwise the only place its called from is from `init` in the service (and passed `nil`) and `get-or-initialize`, where it will always be `keyword`.